cylc/flow/config.py

```diff
@@ -729,20 +729,22 @@ def process_initial_cycle_point(self):
             WorkflowConfigError - if it fails to validate
         """
         orig_icp = self.cfg['scheduling']['initial cycle point']
-        if orig_icp is None:
-            if self.cfg['scheduling']['cycling mode'] == INTEGER_CYCLING_TYPE:
-                orig_icp = '1'
+        if self.cycling_type == INTEGER_CYCLING_TYPE:
```

Comment:
@hjoliver I **think** I saw `cycling_type` in the debugger, and it was `None`? But there was a `cycling mode` in the `cfg['scheduling']` that had the `integer` value. Not sure if that's helpful/important for this issue :+1:

Reply:
`self.cycling_type` is initialized in `WorkflowConfig.__init__()`.  It shouldn't be `None`, but I'll take another look...

Reply:
Unfortunately `self.cfg['scheduling']['cycling mode']` was being used incorrectly here.  In the integer case it is equal to `INTEGER_CYCLING_TYPE` but in the date-time case it is equal to the calendar mode, not to `ISO8601_CYCLING_TYPE`.